[ECE 4.2] IPv6 on AWS tutorial, ECE IPv6 Support, installation instructions, load balancers and OS guides (#5602)
elastic/docs-content: deploy-manage/deploy/cloud-enterprise/ece-ipv6-aws-setup.md

3. **Subnets** → Select each subnet → **Actions** → **Edit subnet settings**
   - Enable **Auto-assign IPv6 address**

4. **Route Tables** → Select the route table for your subnets → **Edit routes**
   - Add route: Destination `::/0` → Target: Your Internet Gateway

5. **Security Groups** → Select or create a security group → **Edit inbound rules**
   - Add the following rules for both IPv4 (`0.0.0.0/0`) and IPv6 (`::/0`):

   | Port | Protocol | Purpose |
   |------|----------|---------|
   | 22 | TCP | SSH access |
   | 443 | TCP | NLB/ALB ingress |
   | 9243 | TCP | ECE proxy (direct access) |
   | 12443 | TCP | Admin Console (direct access) |

## Step 2: Set up EC2 instances [step-vm-launch]

Create one or more RHEL instances to host ECE. Skip this step if ECE is already installed.

::::{note}
This tutorial uses a single-host deployment for simplicity and testing purposes. For production environments, you should deploy multiple hosts across availability zones to ensure high availability and resilience. Refer to [Identify the deployment scenario](/deploy-manage/deploy/cloud-enterprise/identify-deployment-scenario.md) for guidance on recommended architectures.
::::

> 1. **EC2** → **Launch Instance**

Reviewer: **Suggestion: Add note about dual-stack network dependency for egress**

The architecture diagram shows a dual-stack Podman network (`ece-network`), but the creation steps aren't included here. I understand from your earlier comment that you plan to add IPv6/dual-stack support to the "Prepare your environment" docs.

For now, could we add a note here clarifying this dependency? Something like:

```markdown
::::{note}
If you require IPv6 egress (outbound container connections to IPv6 endpoints), you must configure a dual-stack Podman network before installing ECE. Refer to [Prepare your environment](/deploy-manage/deploy/cloud-enterprise/prepare-environment.md) for container runtime IPv6 configuration.
::::
```

This sets the expectation without duplicating content that belongs in the preparation docs.

   - `t3` instances are insufficient for ECE

4. **Key pair**: Select or create a key pair for SSH access

5. **Network settings**:
   - **VPC**: Select your dual-stack VPC
   - **Subnet**: Select a subnet with IPv6 enabled
   - **Auto-assign public IP**: Enable
   - **Security group**: Select your security group with IPv6 rules

6. **Advanced network configuration** (click **Edit**):
   - **IPv4 address**: Assigned by {{aws}}
   - **IPv6 address**: Select "Assigned from CIDR" (your subnet's IPv6 range)
   - **IPv6 prefix**: Leave unchecked

7. **Storage**: Add at least 200GB for `/mnt/data` (ECE data directory)

8. In the UI, select **Launch instance**

### Verify IPv6 outbound connectivity

If you plan to enable IPv6 egress, validate host-level IPv6 outbound connectivity. After the instance launches, SSH into the host and verify:

```bash
# Check IPv6 address is assigned
ip -6 addr show scope global

# Test IPv6 connectivity
ping6 -c 3 ipv6.google.com
```

## Step 3: Prepare the Host and Install ECE [step-install-ece]

Prepare the host according to the official ECE documentation: [Prepare your environment](/deploy-manage/deploy/cloud-enterprise/prepare-environment.md).

> ::::{note}

Reviewer: **Suggestion: Use HTTPS health check with `/_health` endpoint**

TCP health checks only verify the port is open, not that the proxy service is functioning correctly. The ECE proxy exposes a dedicated `/_health` endpoint for health checking.

Suggested change:

```markdown
3. **Health checks**:
   - **Health check protocol**: HTTPS
   - **Health check path**: `/_health`
   - **Health check port**: Traffic port
   - **Healthy threshold**: 3
   - **Unhealthy threshold**: 3
   - **Interval**: 30 seconds
   - **Timeout**: 10 seconds
   - **Success codes**: `200`
```

Also suggest updating the "Verify Proxy NLB" curl commands below to use `/_health`:
```bash
curl -4 -k -s -o /dev/null -w "IPv4: %{http_code}\n" "https://${NLB_DNS}/_health"
curl -6 -k -s -o /dev/null -w "IPv6: %{http_code}\n" "https://${NLB_DNS}/_health"
```
Both should return `200`.

Reviewer: **Note:** This was updated in my latest PR commit (elastic/cloud#153120, commit `b787b62246f`). Your version pulled from an earlier iteration where I had TCP health checks. The change from TCP to HTTPS with `/_health` came from my end-to-end testing - TCP health checks were passing even when the proxy service wasn't fully functional.

**ACM Certificate:**
1. Navigate to **Certificate Manager** → **Request certificate**
2. Request a public certificate for your domain
3. Complete domain validation
4. Note the certificate ARN

**Two subnets in different AZs:**

ALBs require subnets in at least two availability zones. If you only have one subnet, create a second:

1. Go to **VPC → Subnets → Create subnet**
2. Configuration:
   - **VPC**: Select your VPC
   - **Subnet name**: for example, `ece-subnet-2`
   - **Availability Zone**: Select a **different** AZ from your first subnet
   - **IPv4 CIDR**: An unused range (for example, `10.0.32.0/24`)
3. Click **Create subnet**
4. **Add IPv6 to the new subnet:**
   - Select the new subnet → **Actions → Edit IPv6 CIDRs**
   - Add an IPv6 CIDR with a unique suffix (e.g., `2a05:d018:xxx:xx20::/64`)
5. **Associate with route table:**
   - Go to **VPC → Route Tables** → select your route table
   - **Subnet associations → Edit** → Add the new subnet
   - Ensure the route table has routes for `0.0.0.0/0` and `::/0` to the Internet Gateway

### Create the ECE coordinators target group

1. Navigate to **EC2** → **Target Groups** → **Create target group**

> 

Reviewer: **Suggestion: Add RHEL NetworkManager configuration for existing installations**

For existing RHEL 8/9 hosts, assigning an IPv6 address in AWS doesn't automatically configure the OS to use it. NetworkManager often needs explicit configuration.

Suggest adding after "Reconfigure host network interfaces...":

```markdown
#### Configure NetworkManager for IPv6 (RHEL 8/9)

After assigning the IPv6 address in AWS, RHEL may not automatically configure it:

```bash
# Find your connection name
nmcli con show

# Configure IPv6 auto (SLAAC) - replace "System eth0" with your connection name
sudo nmcli con mod "System eth0" ipv6.method auto

# Restart the connection
sudo nmcli con down "System eth0" && sudo nmcli con up "System eth0"

# Verify
ip -6 addr show scope global
ping6 -c 3 ipv6.google.com
```

If `ping6` returns "Network unreachable", verify:
- Your AWS route table has a route for `::/0` pointing to the Internet Gateway
- NetworkManager IPv6 method is set to `auto`
```

2. **Basic configuration**:
   - **Target type**: Instances
   - **Target group name**: `ece-admin-tg-12443`
   - **Protocol**: HTTPS
   - **Port**: 12443
   - **IP address type**: IPv4
   - **VPC**: Select your dual-stack VPC

3. **Health checks**:
   - **Health check protocol**: HTTPS
   - **Health check path**: `/`
   - **Health check port**: Traffic port
   - **Healthy threshold**: 2
   - **Unhealthy threshold**: 2

>    - **Timeout**: 5 seconds

Reviewer: **Suggestion: Add note explaining why container recreation is needed**

This helps users understand what the Support guidance will involve:

```markdown
::::{note}
After updating the proxy configuration via the Container Sets API, running containers retain their previous network state. For the configuration changes to take effect on containers with exposed ports (such as the proxy on port 9243), those containers must be recreated. The Support team can guide you through this process safely.
::::
```

I've created a tracking issue for a KB article with the detailed procedure: https://github.com/elastic/cp-hosted-team/issues/4183

#### ECE requirements

| Requirement | IPv6 ingress | IPv6 egress | Notes |
|-------------|--------------|--------------|-------|
| Proxy Protocol v2 support in ECE proxies | ✔ | | Required for IPv6 ingress through {{aws}} Network Load Balancers (NLB) for deployment traffic |
| IPv6 interfaces on ECE hosts | | ✔ | Required for outbound IPv6 connectivity |
| Container runtime with IPv6 support (Podman or Docker) | | ✔ | Required for outbound IPv6 connectivity |

#### {{aws}} requirements

| Component | Requirement |
|-----------|-------------|
| **VPC** | An associated IPv6 CIDR block (Amazon-provided or BYOIP) |
| **Subnets** | Dual-stack subnets (IPv4 + IPv6) in at least two Availability Zones (required for Application Load Balancers) |
| **Internet Gateway** | Attached to the VPC with routes for both `0.0.0.0/0` and `::/0` |
| **Route Tables** | IPv4 and IPv6 routes to the Internet Gateway |
| **Security Groups** | Inbound rules allowing traffic on ports `443`, `9243`, and `12443` for both `0.0.0.0/0` and `::/0` |
| **EC2 instances** | ECE hosts running in dual-stack subnets (required only for IPv6 egress) |

> 

Reviewer: **Suggestion: Add detailed EC2 IPv6 address assignment steps**

The AWS requirements table mentions EC2 instances need to be in dual-stack subnets, but existing instances don't automatically get IPv6 addresses. Suggest adding this section before "IPv6 ingress in existing IPv4 environments":

```markdown
### Assigning IPv6 addresses to existing EC2 instances

Existing EC2 instances do not automatically receive IPv6 addresses when you enable IPv6 on their subnet. You must assign them manually.

**For each ECE host:**

1. Go to **EC2 → Instances → Select your instance**
2. **Actions → Networking → Manage IP addresses**
3. Expand the network interface section
4. Under **IPv6 addresses**, click **Assign new IP address**
5. Click **Assign** to auto-assign an IPv6 address from your subnet's range
6. Click **Save**

Alternatively, enable auto-assignment for future instances:
- **VPC → Subnets → Select subnet → Actions → Edit subnet settings**
- Enable **Auto-assign IPv6 address**
```

Author: Adding it, great catch!

4. Optionally, configure a dual-stack ALB for admin console UI traffic. Refer to [Step 5: Create the control plane ALB - Optional](#step-alb).

### IPv6 egress in existing IPv4 environments

To support IPv6 egress in an existing IPv4 ECE environment, you must update both host networking and container networking on every ECE host:

1. Reconfigure host network interfaces so each ECE host has working dual-stack connectivity.
2. Reconfigure Podman or Docker bridge/network settings to enable IPv6 for container traffic.

Because these are host-level networking changes, the recommended approach is to reinstall or rebuild ECE hosts one by one, following the official host maintenance procedure: [Perform ECE hosts maintenance](/deploy-manage/maintenance/ece/perform-ece-hosts-maintenance.md).

This rolling approach reduces platform risk and helps preserve service availability while introducing IPv6 egress support.

After completing these changes, refer to [Step 6: Verify IPv6 egress from containers](#step-ipv6-egress) to validate outbound IPv6 connectivity from ECE workloads.

> 

Reviewer: **Suggestion: Add Podman network commands for IPv6 egress on existing installations**

The current text says to "reconfigure Podman or Docker bridge/network settings" but doesn't show how. Suggest adding concrete commands:

```markdown
#### Configure dual-stack Podman network for IPv6 egress

Create the dual-stack network and connect existing containers:

```bash
# Create the dual-stack network
sudo podman network create \
    --subnet 10.89.0.0/24 \
    --subnet fd00:10:89::/64 \
    --ipv6 \
    ece-network

# Set as default for future containers
sudo tee -a /etc/containers/containers.conf > /dev/null <<EOF

[network]
default_network = "ece-network"
EOF

# Connect existing containers to the new network (for IPv6 egress)
for container in $(sudo podman ps -q); do
    sudo podman network connect ece-network "$container" 2>/dev/null || true
done
```

::::{note}
The `podman network connect` command enables IPv6 **egress** (outbound connections) for existing containers. However, it does not update iptables rules for **ingress** (inbound connections). For IPv6 ingress, containers with exposed ports may need to be recreated.
::::

Verify a container has IPv6 connectivity:

```bash
sudo podman exec <container_id> curl -6 -s -o /dev/null -w "%{http_code}" https://ipv6.google.com
```
```

This could alternatively go in the "Prepare your environment" docs as you mentioned - let me know your preference.

Author: @brendandw : 

> The current text says to "reconfigure Podman or Docker bridge/network settings" but doesn't show how.

It was clearly a "placeholder" so we can decide what to do :) Thanks for sharing that procedure!

One thing that we probably must do is to ensure our "OS host preparation" official documents include the necessary steps to prepare a dual-stack host (for both docker and podman based hosts), for new installations.

Would they be similar to the ones you have proposed or would a new / uninstalled host would be configured with a dual-stack default podman network instead of creating a custom `ece-network` network?

For the procedure you are suggesting it looks really nice and smooth, but I'd like you to double check some concerns and potential risks to be totally sure this is ok:

- With the provided steps we are tweaking podman networks in a way that probably a new system won't have, although maybe that's fine.

- We are also adding the containers to an extra net (`ece-network`), besides the original /default bridge network they alrady belong and are connected to, and future containers would belong to `ece-network` only probably. I don't know the implications of that in future ECE upgrades, existing stunnels and things like that.

- The text saying that "containers with exposed ports may need to be recreated." should be clarified a bit more, as for ingress we are actually suggesting to use the original ipv4 adress of the hosts, so I don't think ingress should have any impact except if the user wants ingress towards the ipv6 address to the host (i've added an extra sentence in the note for your consideration).

Of course if you are confident about this, including upgrades, maintenance and support of a production system after these changes then I'm totally ok with it, as the process looks really easy and smooth.

An alternative approach would be telling users to reinstall ECE hosts one by one, enabling IPv6 at host level (with the EC2 and networkmanager example) and then reconfiguring docker per the official doc (the one that we have to update anyway), but of course this would take much longer in a prod system than your proposal.

Reviewer: Thanks Eduardo! We discussed this in more detail on Slack - the short version is:

 - You're right that `containers.conf` setting` ece-network` as default means new containers after upgrades will use it automatically
 - Your concern about mixed subnets (old containers on both networks, new ones only on `ece-network`) is valid - probably fine in practice since ECE uses host IPs for inter-container communication, but it does leave things inconsistent
 - We agreed to keep the procedure but add a note recommending reinstalling ECE hosts one by one via host
    maintenance docs as the safest approach for production

You mentioned you'd test an ECE upgrade on a 3-host lab after applying the procedure - that would be really valuable to confirm how containers end up. Let me know what you find!

If you plan to follow this path, contact [Elastic Support](/troubleshoot/index.md#contact-us) for guidance.

#### Verify ECE proxies configuration

After reconfiguring the proxies, verify that the proxy container includes the required Proxy Protocol v2 environment variables.
Run the following command on each ECE host with the proxy role:

```bash
sudo podman exec $(sudo podman ps --format '{{.Names}}' | grep frc-proxies-proxyv2) env | grep CLOUD_HTTP_PROXY_PROTO
```

Expected output includes both variables:

```
CLOUD_HTTP_PROXY_PROTO_VERSION=2
CLOUD_HTTP_PROXY_PROTO_LENIENT=true
```

## Related Documentation

- [ECE Hardware requirements](/deploy-manage/deploy/cloud-enterprise/ece-hardware-prereq.md)
- [Prepare your environment](/deploy-manage/deploy/cloud-enterprise/prepare-environment.md)
- [Install ECE](/deploy-manage/deploy/cloud-enterprise/install.md)
- [ECE installation procedures](/deploy-manage/deploy/cloud-enterprise/install-ece-procedures.md) (memory settings for different deployment sizes)
- [ECE Load balancers](/deploy-manage/deploy/cloud-enterprise/ece-load-balancers.md)
- [Networking prerequisites](/deploy-manage/deploy/cloud-enterprise/ece-networking-prereq.md)

> - [{{aws}} Elastic Load Balancing documentation](https://docs.aws.amazon.com/elasticloadbalancing/)

Reviewer: **Suggestion: Add final verification checklist**

After all the configuration steps, it would help to have a verification checklist. Suggest adding before "Related Documentation":

```markdown
## Final verification

After completing the configuration for existing installations, verify IPv6 is working end-to-end:

1. **Check target health** in {{aws}} Console: **EC2 → Target Groups → Targets** - all registered targets should show "Healthy"

2. **Test NLB connectivity** from a client with IPv6:
   ```bash
   NLB_DNS="your-nlb-dns-name.elb.region.amazonaws.com"
   
   # Test IPv4
   curl -4 -k -s -o /dev/null -w "IPv4: %{http_code}\n" "https://${NLB_DNS}/_health"
   
   # Test IPv6
   curl -6 -k -s -o /dev/null -w "IPv6: %{http_code}\n" "https://${NLB_DNS}/_health"
   ```
   Both should return `200`.

3. **Verify client IP propagation** by checking proxy logs after making a request:
   ```bash
   sudo podman exec frc-proxies-proxyv2 tail -10 /app/logs/proxy.requests.log | grep client_ip
   ```
   The `client_ip` field should show your actual client IP (IPv4 or IPv6), not an internal IP like `10.89.0.1`.
```

Author: final verification steps look great. I'll also check how to ensure they don't feel repetitive considering in each section we have verifications steps focused on each individual goal.

:::::{note}
This tutorial focuses on new environment setups. If you are working with an existing ECE installation, refer to [Appendix: Integrate IPv6 in existing ECE installations](#existing-installations-summary) to understand the required actions and additional infrastructure changes.
:::::

## Overview

This guide covers two distinct IPv6 traffic flows, ingress and egress, which can be configured independently and have different infrastructure implications:

- **IPv6 ingress (client connectivity)**: Enables IPv4 and IPv6 clients to access ECE endpoints through {{aws}} load balancers. When combined with [IP filtering rules](/deploy-manage/security/ip-filtering-ece.md), this also lets you apply network security policies to IPv6 client addresses. This includes:
  - **Deployment traffic**: Client access to {{es}} and {{kib}} through a dual-stack Network Load Balancer (NLB) and [Proxy Protocol v2](https://docs.aws.amazon.com/elasticloadbalancing/latest/network/edit-target-group-attributes.html#proxy-protocol) used for [client IP propagation](https://docs.aws.amazon.com/elasticloadbalancing/latest/network/edit-target-group-attributes.html#client-ip-preservation).

>   - **Control plane traffic**: Optional access to the ECE Admin Console through a dual-stack Application Load Balancer (ALB).

Reviewer 2: Terminology: `ece-ipv6-support.md` says "Cloud UI" throughout, while this tutorial and the troubleshooting page say "Admin Console". The docs standard is Cloud UI. Can we align, or at least introduce one as an alias on first use?

Reviewer 3: fixed in https://github.com/elastic/docs-content/pull/5602/changes/9438ee99ac0d5724885128e5f1a4140be460d807

## Configuration steps

:::::{stepper}

::::{step} Set up {{aws}} infrastructure
:anchor: step-vpc

Follow these steps in the {{aws}} console to create a new VPC or configure an existing one to enable IPv6 support:

1. **VPC** → **Your VPCs** → Select your VPC (or create new) → **Actions** → **Edit CIDRs**
   - Click **Add new IPv6 CIDR**
   - Select **Amazon-provided IPv6 CIDR block**
   - Click **Select CIDR**

2. **Subnets** → Select each subnet → **Actions** → **Edit IPv6 CIDRs**
   - Click **Add IPv6 CIDR**
   - Enter a unique subnet suffix (for example, `00`, `01`, `02`)
   - This creates a `/64` IPv6 subnet from your VPC's `/56` block

3. **Subnets** → Select each subnet → **Actions** → **Edit subnet settings**
   - Enable **Auto-assign IPv6 address**

4. **Route Tables** → Select the route table for your subnets → **Edit routes**
   - Add route: Destination `::/0` → Target: Your Internet Gateway

5. **Security Groups** → Select or create a security group → **Edit inbound rules**

>    - Add the following rules for both IPv4 (`0.0.0.0/0`) and IPv6 (`::/0`). These rules cover basic external access for a single-node test environment. For production multi-node deployments, refer to [Networking prerequisites](./ece-networking-prereq.md) for the full list of required ports, including inter-node traffic.

Reviewer 2: These rules open ports 9243 and 12443 to `0.0.0.0/0` and `::/0`, which bypasses the load balancers entirely, and people copy these tables into production as-is.

Can we scope those two rules to the load balancer security group or the admin's CIDR? If they're intentional for initial testing, a warning would help, something like: "the direct-access rules for 9243 and 12443 are for initial testing only; once the load balancers are verified, restrict these ports to the load balancer security group."

Reviewer 3: attempted a fix in https://github.com/elastic/docs-content/pull/5602/changes/9172dfb28bf1d82ad6731710d345608187599e3d

   - Avoid marketplace variants, SQL Server editions, or third-party repackaged images

3. **Instance type**:
   - Select an instance with at least 32GB RAM (for example, `r5.xlarge`, `m5.2xlarge`)
   - `t3` instances are insufficient for ECE

4. **Key pair**: Select or create a key pair for SSH access

5. **Network settings**:
   - **VPC**: Select your dual-stack VPC
   - **Subnet**: Select a subnet with IPv6 enabled
   - **Auto-assign public IP**: Enable
   - **Security group**: Select your security group with IPv6 rules

6. **Advanced network configuration** (click **Edit**):
   - **IPv4 address**: Assigned by {{aws}}
   - **IPv6 address**: Select "Assigned from CIDR" (your subnet's IPv6 range)
   - **IPv6 prefix**: Leave unchecked

7. **Storage**: Add at least 200GB for `/mnt/data` (ECE data directory)

8. In the UI, select **Launch instance**

### Verify IPv6 outbound connectivity

If you plan to enable IPv6 egress, validate host-level IPv6 outbound connectivity. After the instance launches, SSH into the host and verify:

```bash
# Check IPv6 address is assigned
ip -6 addr show scope global

# Test IPv6 connectivity
ping6 -c 3 ipv6.google.com
```

::::

> ::::{step} Prepare the Host and Install ECE

Reviewer 2: Sentence case per the style guide.

```suggestion
::::{step} Prepare the host and install ECE
```

    - **IP address type**: IPv4
    - **VPC**: Select your dual-stack VPC

3. **Health checks**:
    - **Health check protocol**: HTTPS
    - **Health check path**: `/_health`
    - **Health check port**: Traffic port
    - **Healthy threshold**: 3
    - **Unhealthy threshold**: 3
    - **Interval**: 30 seconds
    - **Timeout**: 10 seconds
    - **Success codes**: `200`

4. Click **Next**

5. **Register targets**:
    - Select your ECE instance (select all your [ECE proxies](/deploy-manage/deploy/cloud-enterprise/ece-roles.md) in a multi-node environment)
    - **Port**: `9243`
    - Click **Include as pending below**
    - Click **Create target group**

6. **Enable Proxy Protocol v2**:
    - Select the target group → **Attributes** tab → **Edit**
    - Enable **Proxy protocol v2**
    - Click **Save changes**

### Create the dual-stack NLB

1. Navigate to **EC2** → **Load Balancers** → **Create Load Balancer**

2. Select **Network Load Balancer**

3. **Basic configuration**:
    - **Name**: `ece-proxy-nlb`
    - **Scheme**: Internet-facing
    - **IP address type**: **Dualstack** (important - not IPv4)

4. **Network mapping**:
    - **VPC**: Select your dual-stack VPC

>     - **Availability Zones**: Select the same subnet where your ECE instance is running (the NLB must be in the same subnet as the target instance)

Reviewer 2: Same-subnet isn't an AWS requirement: the NLB needs a subnet in the same availability zone as the target (same VPC), not the same subnet. As written this will confuse multi-node setups where proxies span AZs.

```suggestion
    - **Availability Zones**: Enable the availability zones where your ECE proxies run. The NLB needs a subnet in each AZ that contains a target instance.
```

Reviewer 3: fixed in https://github.com/elastic/docs-content/pull/5602/changes/1a5e3616046a07b16d5c967f79a2de3f16a6e3a3

:anchor: step-alb

If you need IPv6 access to the ECE Admin Console, create an Application Load Balancer (ALB).

This configuration follows the requirements described in the [ECE load balancers](/deploy-manage/deploy/cloud-enterprise/ece-load-balancers.md) documentation for admin console traffic.

### ALB prerequisites

Application Load Balancers require an ACM certificate and subnets in at least two availability zones.

**ACM Certificate:**
1. Navigate to **Certificate Manager** → **Request certificate**
2. Request a public certificate for your domain
3. Complete domain validation
4. Note the certificate ARN

**Two subnets in different AZs:**

ALBs require subnets in at least two availability zones. If you only have one subnet, create a second:

1. Go to **VPC → Subnets → Create subnet**
2. Configuration:
   - **VPC**: Select your VPC
   - **Subnet name**: for example, `ece-subnet-2`
   - **Availability Zone**: Select a **different** AZ from your first subnet
   - **IPv4 CIDR**: An unused range (for example, `10.0.32.0/24`)
3. Click **Create subnet**
4. **Add IPv6 to the new subnet:**
   - Select the new subnet → **Actions → Edit IPv6 CIDRs**

>    - Add an IPv6 CIDR with a unique suffix (for example, `2a05:d018:xxx:xx20::/64`)

Reviewer 2: The IPv6 CIDR examples switch style mid-document: the ingress step uses suffix form (`00`, `01`) while this appendix uses a realistic prefix. Worth picking one convention.

Reviewer 3: attempted fix https://github.com/elastic/docs-content/pull/5602/changes/974fd444670918647723bd40ed2e864f9e600bda

    ```

    A response of `200` confirms IPv6 egress is working. You can also test from {{kib}} containers using the same approach.

    :::{dropdown} Optional: test IPv6 egress using {{es}} {{watcher}}
    Run this from any existing deployment to perform a connectivity test using [{{watcher}}](/explore-analyze/alerting/watcher.md):

    ```json
    POST _watcher/watch/_execute
    {
      "watch": {
        "trigger": { "schedule": { "interval": "1h" } },
        "input": {
          "http": {
            "request": {
              "scheme": "https",
              "host": "ipv6.google.com",
              "port": 443,
              "method": "head",
              "path": "/"
            }
          }
        },
        "actions": {
          "log": {
            "logging": { "text": "IPv6 test - Status: {{ctx.payload.status}}" }
          }
        }
      }
    }
    ```
    :::

::::

:::::

## Troubleshooting

For troubleshooting and verification commands specific to IPv6 integration on {{aws}}, refer to [Troubleshoot IPv6 integration for ECE on {{aws}}](/troubleshoot/deployments/cloud-enterprise/troubleshoot-ipv6-aws-integration.md).

> ## Related Documentation

Reviewer 2: Sentence case. (The Proxy Protocol page uses just "Related"; either works, but worth aligning the two pages.)

```suggestion
## Related documentation
```

    ```bash
    # List connections and identify the active one
    nmcli con show

    # Example for a connection named "System eth0"
    sudo nmcli con mod "System eth0" ipv6.method auto

    # Restart the connection
    sudo nmcli con down "System eth0" && sudo nmcli con up "System eth0"

    # Verify host IPv6 connectivity
    ip -6 addr show scope global
    ping6 -c 3 ipv6.google.com
    ```

    If `ping6` returns `Network unreachable`, verify:

    - Route tables include `::/0` to the Internet Gateway.
    - NetworkManager uses `ipv6.method auto` on the active connection.

3. Configure Podman dual-stack network for IPv6

    On existing ECE hosts that were prepared without IPv6, create a dual-stack network in Podman, and configure it as the default network for future containers:

    ```bash
    # Create a dual-stack network
    sudo podman network create \
      --subnet 10.89.0.0/24 \
      --subnet fd00:10:89::/64 \
      --ipv6 \
      ece-network

    # Set as default for future containers
    sudo tee -a /etc/containers/containers.conf > /dev/null <<'EOF'
    [network]
    default_network = "ece-network"

>     EOF

Reviewer 4: This `tee -a` creates a second `[network]` section. The new-host RHEL guide says to merge `default_network` into the existing `[network]` block (`network_backend="cni"` on RHEL 9 / Rocky 9) rather than duplicating the section.

Please use the same merge instructions here so existing installs don’t end up with an invalid or ambiguous `containers.conf`.

<small>*(AI-authored via Cursor 🤖)*</small>

Reviewer 3: fixed in https://github.com/elastic/docs-content/pull/5602/changes/0fa83b1527652dddcc59d5d3a9f643ce3ee396f6
